src/TapeRoller.ts

```diff
@@ -0,0 +1,253 @@
+import * as fs from 'fs';
+import { resolve } from 'path';
+import { promisify } from 'util';
+import streamReplace from 'stream-replace';
+import { Homefront } from 'homefront';
+import { exec, spawn } from 'child_process';
+import rimraf from 'rimraf';
+import mkdirp from 'mkdirp';
+import glob from 'glob';
+
+export type ConfigType = Partial<{
+  sourceDirectory: string,
+  targetDirectory: string,
+  sourceFile: string,
+  targetFile: string,
+}>;
+
+export type ModifyType = Partial<{
+  append: string,
+  prepend: string,
+  replace: string,
+  custom: Function,
+}>;
+
+export type StreamType = fs.ReadStream | fs.WriteStream;
+
+export class TapeRoller {
+  private readonly sourceDirectory?: string;
+
+  private sourceFile?: string;
+
+  private readonly targetDirectory?: string;
+
+  private targetFile?: string;
+
+  private stream?: StreamType;
+
+  constructor ({ sourceDirectory, targetDirectory, sourceFile, targetFile }: ConfigType = {}) {
+    this.sourceDirectory = sourceDirectory;
+    this.targetDirectory = targetDirectory;
+    this.sourceFile      = sourceFile;
+    this.targetFile      = targetFile;
+  }
+
+  private static resolvePath (dir?: string, file?: string): string {
+    return resolve(process.cwd(), dir || '', file || '');
+  }
+
+  public static async fromGlob (pattern: string, options?: { [key: string]: any }, targetDirectory?: string): Promise<TapeRoller[]> {
+    const asyncGlob = promisify(glob);
+    const files     = await asyncGlob(pattern, options);
+
+    return files.map(sourceFile => new TapeRoller({ targetDirectory, sourceFile }));
+  }
+
+  public rename (oldPath: string, newPath: string): this {
+    fs.renameSync(oldPath, newPath);
```

Comment:
Why sync? Can easily be async with an await I'd say.

Reply:
Since renaming is, most likely, the last step (if used at all) there's no real reason for it. It was a last minute change I've made when I wrongfully assumed that someone might need to chain another method after `rename()` relying on the file to be renamed.

Reply:
Coolio. So you'll make it async?

Reply:
Absolutely I do